src/CommunityToolkit.Maui.MediaElement/Views/MediaManager.android.cs

```diff
@@ -181,35 +187,77 @@ or PlaybackStateCompat.StateSkippingToQueueItem
 	/// <returns>The platform native counterpart of <see cref="MediaElement"/>.</returns>
 	/// <exception cref="NullReferenceException">Thrown when <see cref="Android.Content.Context"/> is <see langword="null"/> or when the platform view could not be created.</exception>
 	[MemberNotNull(nameof(PlayerView))]
-	[MemberNotNull(nameof(uiUpdateReceiver))]
-	[MemberNotNull(nameof(checkPermissionsTask))]
-	[MemberNotNull(nameof(mediaSessionConnector))]
-	[MemberNotNull(nameof(mediaControllerCompat))]
-	public (PlatformMediaElement platformView, StyledPlayerView PlayerView) CreatePlatformView()
+	public (PlatformMediaElement platformView, PlayerView PlayerView) CreatePlatformView()
 	{
 		ArgumentNullException.ThrowIfNull(MauiContext.Context);
-		Player = new IExoPlayer.Builder(MauiContext.Context).Build() ?? throw new NullReferenceException();
+		Player = new ExoPlayerBuilder(MauiContext.Context).Build() ?? throw new InvalidOperationException("Player cannot be null");
 		Player.AddListener(this);
-		InitializeMediaSession();
 
-		PlayerView = new StyledPlayerView(MauiContext.Context)
+		Player.SetHandleAudioBecomingNoisy(true);
+		var audioAttributes = new AudioAttributes.Builder();
+		audioAttributes.SetUsage(C.UsageMedia);
+		audioAttributes.SetContentType(C.DataTypeMedia);
+		Player.SetAudioAttributes(audioAttributes.Build(), true);
+
+		PlayerView = new PlayerView(MauiContext.Context)
 		{
 			Player = Player,
 			UseController = false,
 			ControllerAutoShow = false,
 			LayoutParameters = new RelativeLayout.LayoutParams(ViewGroup.LayoutParams.MatchParent, ViewGroup.LayoutParams.MatchParent)
 		};
 
-		checkPermissionsTask = CheckAndRequestForegroundPermission(checkPermissionSourceToken.Token);
+		string RandomId = Convert.ToBase64String(Guid.NewGuid().ToByteArray())[..8];
+		var mediaSessionWRandomId = new AndroidX.Media3.Session.MediaSession.Builder(Platform.AppContext, Player);
+		mediaSessionWRandomId.SetId(RandomId);
+		session ??= mediaSessionWRandomId.Build();
+
+		if (Build.VERSION.SdkInt < BuildVersionCodes.Tiramisu)
+		{
+			uiUpdateReceiver ??= new UIUpdateReceiver();
+			IntentFilter intentFilter = new(MediaControlsService.ACTION_UPDATE_PLAYER);
+			var flags = ContextCompat.ReceiverNotExported;
+			ContextCompat.RegisterReceiver(Platform.AppContext, uiUpdateReceiver, intentFilter, flags);
+
+			notificationService ??= Microsoft.Maui.Controls.Application.Current?.MainPage?.Handler?.MauiContext?.Services?.GetService<INotificationService>();
```

Comment:
this isn't the best way to get the MainPage anymore... For more info you can see this [PR](https://github.com/dotnet/maui/pull/23692)

I think we can implement [this helper](https://github.com/davidortinau/WeatherTwentyOne/blob/main/src/WeatherTwentyOne/Services/ServiceExtensions.cs) class to make it easier to get services